code/modules/tables/tables.dm

```diff
@@ -361,19 +369,37 @@
 			if (tabledirs & turn(dir,-90))
 				type += "+"
 
-		icon_state = "flip[type]"
 		if(material)
-			var/image/I = image(icon, "[material.icon_base]_flip[type]")
-			I.color = material.icon_colour
+			if(material.table_icon)
+				if(reinforced && ("reinf_[material.name]_flip[type]" in icon_states(material.table_icon)))
+					I = image(material.table_icon, "reinf_[material.name]_flip[type]")
+				else
+					I = image(material.table_icon, "[material.name]_flip[type]")
+			else
+				if(reinforced && ("reinf_[material.icon_base]_flip[type]" in icon_states(icon)))
+					I = image(icon, "reinf_[material.icon_base]_flip[type]")
+				else
+					I = image(icon, "[material.icon_base]_flip[type]")
+				if(material.icon_colour)
+					I.color = material.icon_colour
 			add_overlay(I)
-			if(material_alteration & MATERIAL_ALTERATION_NAME)
-				name = "[material.display_name] table"
+			if(material.display_name)
+				if(material.display_name == "comfy")
+					name = "fancy table"
+				else
+					name = "[material.display_name] table"
```

Comment:
Is there any particular reason why this is done in the icon update proc, instead of Initialize()?

Reply:
cause you can remove the material and re-add it again - it's not like a chair where when you deconstruct it and construct it again it despawns and respawns it